Svc/FpySequencer/FpySequencerDirectives.cpp

```diff
@@ -418,6 +426,47 @@ Signal FpySequencer::storeTlmVal_directiveHandler(const FpySequencer_StoreTlmVal
     return Signal::stmtResponse_success;
 }
 
+Signal FpySequencer::pushTlmValAndTime_directiveHandler(const FpySequencer_PushTlmValAndTimeDirective& directive,
+                                                        DirectiveError& error) {
+    if (!this->isConnected_getTlmChan_OutputPort(0)) {
+        error = DirectiveError::TLM_GET_NOT_CONNECTED;
+        return Signal::stmtResponse_failure;
+    }
+
+    Fw::Time tlmTime;
+    Fw::TlmBuffer tlmValue;
+    Fw::TlmValid valid = this->getTlmChan_out(0, directive.get_chanId(), tlmTime, tlmValue);
+
+    if (valid != Fw::TlmValid::VALID) {
+        // could not find this tlm chan
+        error = DirectiveError::TLM_CHAN_NOT_FOUND;
+        return Signal::stmtResponse_failure;
+    }
+
+    U8 tlmTimeBuf[Fw::Time::SERIALIZED_SIZE];
+    Fw::ExternalSerializeBuffer timeEsb(tlmTimeBuf, Fw::Time::SERIALIZED_SIZE);
+    Fw::SerializeStatus stat = timeEsb.serializeFrom(tlmTime);
+
+    // coding error if this failed, we should have enough space
+    FW_ASSERT(stat == Fw::SerializeStatus::FW_SERIALIZE_OK, static_cast<FwAssertArgType>(stat));
+
+    // check that our stack won't overflow if we put both val and time on it
+    // cast to u64 first so that we don't get u32 overflow
+    if (static_cast<U64>(this->m_runtime.stackSize) + tlmValue.getBuffLength() + timeEsb.getBuffLength() >
```

Comment:
I prefer to write these as subtractions, because then you don't need to deal with casting.

```
// Make sure the operation will ever be valid
FW_ASSERT(tlmValue.getBuffLength() + timeEsb.getBuffLength() < Fpy::MAX_STACK_SIZE);
if (Fpy::MAX_STACK_SIZE - lmValue.getBuffLength() - timeEsb.getBuffLength() < this->m_runtime.stackSize) { // overflow}